lib/apiservers/engine/backends/cache/image_cache.go

```diff
@@ -209,3 +174,43 @@ func (c *ImageCache) GetImageByNamed(named reference.Named) (*metadata.ImageConf
 
 	return config, nil
 }
+
+// AddImage adds an image to the image cache
+func (c *ImageCache) AddImage(imageConfig *metadata.ImageConfig) {
+	c.m.Lock()
+	defer c.m.Unlock()
+	var imageID string
+
+	// Don't assume the image id in image has "sha256:<id> as format.  We store it in
+	// this fomat to make it easier to lookup by digest
+	if strings.HasPrefix(imageConfig.ImageID, "sha") {
```

Comment:
It can, see [here](https://github.com/docker/docker/blob/3d13fddd2bc4d679f0eaa68b0be877e5a816ad53/vendor/src/github.com/docker/distribution/digest/digester.go#L16).